### DISCUSSION TITLE
Problem with X-axis  scrolling with custom datafeed

### DISCUSSION THREAD

**Discussion by Yachter** [highfestiva/finplot#265] [Q&A]
Hello,

I'm having trouble with scaling after using my data.

complicated2.py:
http://www.fileconvoy.com/dfl.php?id=gd499040d809ca23910003808265806a0b8efaa4d75

Also, I tried to adopt other examples, like embed.py, but get similar problems:

http://www.fileconvoy.com/dfl.php?id=gf32f7d40d37ef69c1000380828c1570d00086ac041

Appreciate any help.

**Comment by highfestiva** [comment]
You have your timestamps in the reversed order. Do `df = df.sort_index()` to fix.

**Reply by Yachter** [reply]
Thank you very much.  One question more:

How to turn off both lines for Stoch indicator ? Probably it's some bug in complicated.py example.

**Comment by highfestiva** [comment]
Write some code of your own, as of now that viewbox can be hidden/shown, no further detail is available in this example. It is not a trading tool, it is an example.

**Comment by Yachter** [comment]
![Stoch](https://user-images.githubusercontent.com/24828886/138163322-3507e557-b244-45c3-86d4-446940bf4997.jpg)

**Reply by highfestiva** [reply]
How to reproduce?

**Reply by Yachter** [reply]
Just run the original complicated.py, select 'moar indicators', and turn off RSI and Stoch as shown in the picture.

Btw, you've made a really brilliant financial charting library, nothing comparable to existing in the python world, trust me & congrats, man. I'm using app shots in my postgraduate research reporting on financial markets, and really thankful. It would be valuable if you consider some of my suggestions and remark to make the app better.

**Reply by Yachter** [reply]
There're some of the things that are unclear to me. For example, in my application, I'm trying to adjust the range of the Y-scale to fit
a new rectangle, because it is outside of data's value range, so make that rectangle visible:

`y_min, y_max = ax.axes['left']['item'].range

 ax.axes['left']['item'].setRange(y_min, (rectangle_Y_max * 1.10))

fplt.refresh()

rect = fplt.add_rect((rect_time_start,rectangle_Y_min), (rect_time_stop, rectangle_Y_max), color='#0f0f27', interactive=False)`

this is code I placed at the end of the callback function of QComboBox, but it does not change the range of main candle chart.

**Reply by highfestiva** [reply]
Change to this: `plots['stoch_s'] = fplt.plot(data['stoch_s'], color='#650', ax=ax_rsi, legend='Stoch sig')` to be able to uncheck visibility.

Try `finplot.set_y_range()` for visibility instead.

**Reply by Yachter** [reply]
Only `finplot.set_y_range()` works, but that's fine. 

So need your advice: is there a callback function for the AutoZoom feature, 
so be able to set the scale and make visible all objects do I need to see?

Another question related to that what is alternative for Y-viewport scale
similar to X-axis `fplt.set_x_pos(xmin, xmax, ax)`  function ?

**Reply by highfestiva** [reply]
No. You might try `ax.vb.set_range(x0, y0, x1, y1)`.
